addons/bus/static/src/js/longpolling_bus.js

```diff
@@ -0,0 +1,252 @@
+        $(window).on("keydown." + this._id, this._onPresence.bind(this));
+        $(window).on("keyup." + this._id, this._onPresence.bind(this));
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        var self = this;
+        this.stopPolling();
+        $(window).off("focus." + this._id);
+        $(window).off("blur." + this._id);
+        $(window).off("unload." + this._id);
+        _.each('click,keydown,keyup'.split(','), function (evtype) {
+            $(window).off(evtype + "." + self._id);
+        });
+    },
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+    /**
+     * Register a new channel to listen on the longpoll (ignore if already
+     * listening on this channel).
+     * Aborts a pending longpoll, in order to re-start another longpoll, so
+     * that we can immediately get notifications on newly registered channel.
+     *
+     * @param  {string} channel
+     */
+    addChannel: function (channel) {
+        if (this._channels.indexOf(channel) === -1) {
+            this._channels.push(channel);
+            if (this._pollRpc) {
+                this._pollRpc.abort();
+            } else {
+                this.startPolling();
+            }
+        }
+    },
+    /**
+     * Unregister a channel from listening on the longpoll.
+     *
+     * Aborts a pending longpoll, in order to re-start another longpoll, so
+     * that we immediately remove ourselves from listening on notifications
+     * on this channel.
+     *
+     * @param  {string} channel
+     */
+    deleteChannel: function (channel) {
+        var index = this._channels.indexOf(channel);
+        if (index !== -1) {
+            this._channels.splice(index, 1);
+            if (this._pollRpc) {
+                this._pollRpc.abort();
+            }
+        }
+    },
+    /**
+     * Tell whether odoo is focused or not
+     *
+     * @returns  {Boolean}
+     */
+    isOdooFocused: function () {
+        return this._isOdooFocused;
+    },
+    /**
+     * Start a long polling, i.e. it continually opens a long poll
+     * connection as long as it is not stopped (@see `stopPolling`)
+     */
+    startPolling: function () {
+        if (!this._isActive) {
+            this._isActive = true;
+            this._poll();
+        }
+    },
+    /**
+     * Stops any started long polling
+     *
+     * Aborts a pending longpoll so that we immediately remove ourselves
+     * from listening on notifications
+     * on this channel.
+     */
+    stopPolling: function () {
+        this._isActive = false;
+        this._channels = [];
+        if (this._pollRpc) {
+            this._pollRpc.abort();
+        }
+    },
+    /**
+     * Add or update an option on the longpoll bus.
+     * Stored options are sent to the server whenever a longpoll is started.
```

Comment:
_"[...] whenever a **longpoll** is started."_ => _"[...] whenever a **poll** is started"_

Reason is that, based on the code:
- **longpoll** is the process of making a poll regularly (every 60 sec.)
- **poll** is the process of making a RPC and keeping it open up up to (60 sec.) is "open".

In this case, options are sent every poll, not every longpoll!